pkg/unidling/api/types.go

```diff
@@ -0,0 +1,20 @@
+package api
+
+import (
+	kextapi "k8s.io/kubernetes/pkg/apis/extensions"
+)
+
+const (
+	IdledAtAnnotation      = "idling.alpha.openshift.io/idled-at"
+	UnidleTargetAnnotation = "idling.alpha.openshift.io/unidle-targets"
+	NeedPodsReason         = "NeedPods"
+)
+
+// RecordedScaleReference is a SubresourceReference to a scale subresource that also
+// has the previous replica count recorded
+type RecordedScaleReference struct {
+	// TODO: update this to a CrossVersionObjectReference once we pull in a Kube version
+	// with an unversioned apis/autoscaling
+	kextapi.SubresourceReference `json:",inline"`
```

Comment:
Concretely, if you rely on `ScaleInterface.Update`, you are broken right here: https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/scale.go#L66 .  While you are likely to use that API, you should avoid having bad data until just before your call.